pkg/controller/build/action.go

```diff
@@ -61,3 +64,14 @@ func (action *baseAction) InjectLogger(log log.Logger) {
 func (action *baseAction) InjectRecorder(recorder record.EventRecorder) {
 	action.recorder = recorder
 }
+
+func (action *baseAction) findCreator(ctx context.Context, build *v1.Build) *corev1.ObjectReference {
+	var kit v1.IntegrationKit
+
+	// Fetch the IntegrationKit instance
+	if err := action.client.Get(ctx, types.NamespacedName{Name: build.Name, Namespace: build.Namespace}, &kit); err == nil {
```

Comment:
For now I wanted to have a non-offensive change to not break the reconcile loop just for logging reasons. I mean the result of this is only used for logging reasons.

Reply:
What about propagating the `camel.apache.org/created.by` annotations to the Build, instead of having to de-reference the owner?

Otherwise, we are biaised toward fail-fast error handling, so we avoid swallowing errors, even for non critical functions.

Reply:
@astefanutti which value should `camel.apache.org/created.by` have for a Build then? the IntegrationKit or the Integration that created the IntegrationKit?

We could avoid the IntegrationKit lookup here completely when `camel.apache.org/created.by` for a Build is the Integration

Reply:
I'd be inclined to report the annotations as is, i.e., pointing to the initiating Integration. WDYT?